### PR TITLE
C++: Add FailedExtractions.ql

### PR DESCRIPTION
**Pull request by igfoo**

### CODE REVIEW

File: cpp/ql/src/Diagnostics/FailedExtractions.ql [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/**
+ * @kind diagnostic
```
Comment (criemen) [inline cpp/ql/src/Diagnostics/FailedExtractions.ql:2]: Per the spec (that was probably updated after you wrote the query), this is missing a name and description.

**Review by criemen — CHANGES_REQUESTED** [review]
One nitpick.
I'm happy to merge this now as a PoC, but I think we should take a look together at the requirements and what they mean for us in the wider context of the epic.

File: cpp/ql/src/Diagnostics/FailedExtractions.ql (hunk 1 above)
Comment (igfoo) [reply cpp/ql/src/Diagnostics/FailedExtractions.ql:4]: Ta, added

**Review by criemen — APPROVED** [review]
LGTM now

**Pull request merged**